lisa/_assets/kmodules/sched_tp/wq.c

```diff
@@ -31,17 +31,33 @@ struct work_item *start_work(worker_t f, int delay, void *data) {
 	item = kmalloc(sizeof(*item), GFP_KERNEL);
 	if (item) {
 		item->f = f;
+		item->cpu = cpu;
 		item->data = data;
 
 		item->__delay = delay;
 		item->__wq = wq;
 		INIT_DELAYED_WORK(&item->__dwork, worker);
 
-		queue_delayed_work(wq, &item->__dwork, delay);
+		if ((unsigned)cpu >= nr_cpu_ids)
```

Comment:
this logic is duplicated with start_work_on, can we make a helper to handle that ?